### ISSUE TITLE
Problem: incremental migrations in extensions

### ISSUE THREAD

**Issue by yrashk** [omnigres/omnigres#711]
They are proving to be more difficult than I was hoping for. We've dealt with function changes by employing sophisticated post-installation comparison procedure (firstly here in this repo and now in [pgpm](https://github.com/postgres-pm/pgpm/blob/master/packages/omnigres/scripts/generate_upgrade.sh)), but there are still issues outstanding:

* Removal of C functions. If we always _incrementally_ migrate that means we still have to expose C functions, even if as [stubs](https://github.com/omnigres/omnigres/pull/707/files#diff-5f58661fd6eb91edd28cb1bf0916ebf1cd5b9887267730ec044631660bb08b92) just to satisfy migrations.
* We go changes for no good reasons even for new installations

Proposed solution: generate upgrades from incremental changes with initial human assistance

I am working on updated view-based migrations for applications, where we observe two schemas and derive the changes asking the user in cases of ambiguities and using this to populate view schemas. The view-based approach does not work for extensions. However, the comparison and interviewing the user should work!

So, basically, what if we wrote extensions as simple .sql files (with includes as we have them now for convenience) but they wouldn't be migrations and then, to prepare proper migrations, we scanned for every commit of the schema that was changed and derive/interview for the changes and save those – to be converted into migrations upon release?

We can even do this in a rolling manner, without having to convert every extension in the repo at once. We can introduce the `schema/` directory convention where we'd put the schema to be scanned. If only the `mirgate` directory is present, then we will use the old method. Until we've completed transition.
